src/modules/client/pages/ni/billing/CreditNotes.vue

```diff
@@ -434,7 +434,10 @@ export default {
     },
     formatPayloadWithSubscription (creditNote, customer) {
       const payload = {};
-      const subscription = customer.subscriptions.find(sub => sub._id === creditNote.subscription);
+      const selectedCustomer = customer._id
+        ? customer
+        : this.customersOptions.find(cus => cus.value === customer) || customer;
```

Comment:
Dans quel cas customer._id n'est pas defini ? Si customer n'est pas dans customerOptions est-ce qu'il ne faudrait pas plutot renvoyer une erreur ? `selectedCustomer.subscriptions.find` risque de casser non ?

Reply:
Il y a deux cas : 
- Soit le payload vient de la creation, dans ce cas, customer est un objectId.
- Soit le payload vient de l'édition, dans ce cas le customer est un objet avec un champ _id 

Dans les deux cas on veut un selectedCustomer sous forme d'objectId.

Reply:
Je suis d'accord avec ton deuxième commentaire par contre, si `this.customersOptions.find(cus => cus.value === customer)` renvoie `undefined`, on doit renvoyer une erreur, je t'ai fait une proposition dans mon commit

Reply:
Top merci ! :) 
Je me demande si en mongoose, il y a un truc du genre : si la variable est un objectId alors on peut faire variable._id et ca renvoie l'objectId aussi. 
Comme on est en front, j'aurais tendnace a dire que ca ne devrait pas poser probleme, mais dans le doute, est-ce qu'il ne faudrait pas plutot verifier si subscription existe vu qu'au final c'est ce champs qui nous interesse ?

Reply:
Il me semble que oui pour ta première remarque
Pour la deuxième remarque, il n'y a aucune raison valable pour laquelle `subscription` n'existerait pas ici. Si c'est le cas, le bug vient d'ailleurs, je ne vois pas pourquoi on remonterait l'erreur ici 🤷‍♂️

Reply:
Pardon je me suis mal exprimee. Je me disais qu'a la place de verifier customer._id pour savoir si customer est un objectId ou un objet, on pourrait verifier customer.subscription pour deux raisons: 
- cette histoire de ._id qui fonctionne en back (en front je ne sais pas)
- enfait pour la suite on a uniquement besoin de subscription, donc c'est vraiment cette info qui nous importe.

Reply:
ah bien vu ! C'est fait